fdbclient/FileBackupAgent.actor.cpp

```diff
@@ -458,6 +468,309 @@ Value makePadding(int size) {
 	return pad.substr(0, size);
 }
 
+struct IRangeFileWriter {
+public:
+	virtual Future<Void> padEnd(bool final) = 0;
+
+	virtual Future<Void> writeKV(Key k, Value v) = 0;
+
+	virtual Future<Void> writeKey(Key k) = 0;
+
+	virtual Future<Void> finish() = 0;
+
+	virtual ~IRangeFileWriter() {}
+};
+
+struct SnapshotFileBackupCipherKey {
+	EncryptCipherDomainId encryptDomainId;
+	EncryptCipherBaseKeyId baseCipherId;
+	EncryptCipherRandomSalt salt;
+	StringRef baseCipher;
+
+	static SnapshotFileBackupCipherKey fromBlobCipherKey(Reference<BlobCipherKey> keyRef, Arena& arena) {
```

Comment:
Any reason BlobCipherKey can't be passed around?

Reply:
I will change the class structure